api/server/server.go

```diff
@@ -191,6 +192,7 @@ func (s *Server) createMux() *mux.Router {
 	notFoundHandler := httputils.MakeErrorHandler(pageNotFoundError{})
 	m.HandleFunc(versionMatcher+"/{path:.*}", notFoundHandler)
 	m.NotFoundHandler = notFoundHandler
+	m.MethodNotAllowedHandler = notFoundHandler
```

Comment:
gorilla/mux now uses a separate handler for non-existent routes. This changes it back to the old behavior.

Reply:
Ah! nice find; perhaps worth adding a comment, explaining why we do this?

Reply:
okz